IMDA-BTG/aiverify
# Feature - editable global variables

## Description

This PR is for implementation of  [Issue 58](https://github.com/IMDA-BTG/aiverify/issues/58) - Editable global variables
It also contains a few minor improvements:

- Project Info global variables will be displayed on the panel but will not be editable. Any other user-added variables will be editable.
- Long global variable text values will be shortened with ellipsis suffix on display. Mouse over the text displays tooltip container the full text

## Motivation and Context

Currently global variables cannot be edited. The only way to edit is to delete and re-add it.

## Type of Change


- feat: A new feature










## How to Test

- Create a new project and land on canvas screen. Check that project info details are added as global variables but they cannot be edited
- Add a new variable and use it on a widget (eg: Header widget). Test edit and deletion of the new variable.
- Generate report to test that the correct global variable values appear on the generated report

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x ]  I have tested the changes locally and verified that they work as expected.
- [ x]  I have followed the project's coding conventions and style guidelines.
- [x ]  I have rebased my branch onto the latest commit of the main branch.
- [ x]  I have squashed or reorganized my commits into logical units.
- [x ]  I have performed a self-review of my own code.
- [ x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

![globalVars-edit](https://github.com/IMDA-BTG/aiverify/assets/130328114/a131835b-c157-40b8-aa72-3db38194b17c)


## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
